xdsl/interpreters/eqsat_pdl_interp.py

```diff
@@ -288,6 +313,13 @@ def eclass_union(
 
         if a == b:
             return False
+
+        # Meet the analysis states of the two e-classes
+        for analysis in self.analyses:
+            a_lattice = analysis.get_lattice_element(a.result)
+            b_lattice = analysis.get_lattice_element(b.result)
+            a_lattice.meet(b_lattice)
```

Comment:
Not really this PR, but thoughts on having meet be `__and__` and join be `__or__`?

Reply:
I like this! though will leave it for a follow up (sparse backwards analysis and dense analysis haven't been fully ported either)